.evergreen/retry-with-backoff.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
```

Comment:
Remove this line – this is no longer a standalone script and is expected to always be `source`d by other scripts, so that should be reflected here

Reply:
We need the line for shellcheck, I added a note instead:

```
^-- SC2148 (error): Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
```

Reply:
I'd still consider using 

```suggestion
# shellcheck shell=sh
```

instead, shebangs are really intended for marking scripts as executable on their own. (Fwiw, you could also use this as a standalone script, like Compass does).

Reply:
Thanks!